### PR TITLE
Add scheme for url

### LINKED ISSUE #446: Urls when http / https is missing
When http is missing, it converts the url to be a part of the website ("https://elixir-companies.com/url").

Edit: not just the website url, we need to force users to enter the correct protocol for their website, so we do not have to guess which one works.

### PR DESCRIPTION
**Pull request by gonglexin**
Fix #446

### CODE REVIEW

File: lib/companies_web/views/company_view.ex [hunk 1]
```diff
@@ -16,6 +16,10 @@ defmodule CompaniesWeb.CompanyView do
     |> String.replace("www.", "")
   end
 
+  def url_with_scheme("http://" <> _ = url), do: url
+  def url_with_scheme("https://" <> _ = url), do: url
+  def url_with_scheme(url, scheme \\ "http"), do: "#{scheme}://#{url}"
```
Comment (doomspork) [inline lib/companies_web/views/company_view.ex]: I think we want the default to be `https`, what do you think?

**Comment by doomspork** [comment]
Thank you for the PR @gonglexin! One comment 👍

File: lib/companies_web/views/company_view.ex (hunk 1 above)
Comment (gemantzu) [reply lib/companies_web/views/company_view.ex]: Even one or the other might lead to issues (like a url not been set properly to redirect to one or the other), we maybe want to do something more clever on the long run. It's fine for now in https I guess

Comment (doomspork) [reply lib/companies_web/views/company_view.ex]: I can look into the cloudflare setup to ensure http goes to https.

Comment (gemantzu) [reply lib/companies_web/views/company_view.ex]: But that's the thing, one might have setup their domain to only use one of them (I know, I have done this shit already :blush)

Comment (tajchumber) [reply lib/companies_web/views/company_view.ex]: how about trying to retrieve the endpoint from here?
https://github.com/doomspork/elixir-companies/blob/2872b679c40b468e7ae4c01e0d6c34deb4bc7545/config/prod.exs#L14

Comment (gonglexin) [reply lib/companies_web/views/company_view.ex]: I'm thinking default scheme to https and open another issue for adding scheme required validation for company & job url, what's your opinion?

Comment (doomspork) [reply lib/companies_web/views/company_view.ex]: @gonglexin sounds good to me 👍

**Pull request merged**